tests/unit/infrastructure/s3/s3Service.test.ts

```diff
@@ -0,0 +1,108 @@
+import { S3 } from 'aws-sdk';
+import getFromS3 from '../../../../src/infrastructure/s3/s3Service';
+
+describe('S3 Service', () => {
+  it('passes the expected key to getObject if folder is defined', async () => {
+    const mockS3 = ({} as unknown) as S3;
+    const bucket = 'bucket';
+    const folder = 'folder';
+    const certNumber = 'cert123456';
+    const vin = 'VIN2345AB';
+    const mockPromise = jest
+      .fn()
+      .mockReturnValue(Promise.resolve({ Body: 'Success!', ContentType: 'application/octet-stream' }));
+    const mockGetObject = jest.fn().mockReturnValue({ promise: mockPromise });
+
+    mockS3.getObject = mockGetObject;
+    await getFromS3(mockS3, bucket, folder, certNumber, vin).catch(() => {});
+
+    const firstCall = mockGetObject.mock.calls[0] as S3.GetObjectRequest[];
+    const firstArg = firstCall[0];
+
+    expect(firstArg.Key).toEqual(`${folder}/${certNumber}_${vin}.pdf`);
+  });
+
+  it('passes the expected key to getObject if folder is undefined', async () => {
+    const mockS3 = ({} as unknown) as S3;
+    const bucket = 'bucket';
+    const folder = undefined;
+    const certNumber = 'cert123456';
+    const vin = 'VIN2345AB';
+    const mockPromise = jest
```

Comment:
Will leave up to you but we seem to be rewriting the promise mock quite a lot so we could have made the code drier using a `beforeEach()` block and [mockImplementation](https://jestjs.io/docs/mock-functions#mock-implementations) instead allowing us to redefine it where needed. It seems our base test case have the following promise returned `{ Body: 'Success!', ContentType: 'application/octet-stream' }));`